[GLib] Re-enable bwrap sandbox in containers when supported

TingPing opened the pull request:
#### 4c39f38757287198c49091ee12fc548cc9c212d6
<pre>
[GLib] Re-enable bwrap sandbox in containers when supported
<a href="https://bugs.webkit.org/show_bug.cgi?id=256917">https://bugs.webkit.org/show_bug.cgi?id=256917</a>

Reviewed by Michael Catanzaro.

This detects if bwrap actually works inside of a container instead
of always disabling it.

* Source/WTF/wtf/glib/Sandbox.cpp:
(WTF::isInsideUnsupportedContainer):
(WTF::isInsideContainer): Deleted.
* Source/WTF/wtf/glib/Sandbox.h:
* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):

Canonical link: <a href="https://commits.webkit.org/264196@main">https://commits.webkit.org/264196@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5dea3830e2daecc45e9c4d0ae80dcc7eb67772b3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6937 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7170 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7353 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8544 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7178 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8431 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7108 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10086 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7059 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7740 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6387 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8644 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5123 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6299 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14085 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/5857 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6799 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6389 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9217 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6515 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6868 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5630 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/7074 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6244 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1655 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1658 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10430 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/7270 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6627 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1793 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WTF/wtf/glib/Sandbox.cpp:
```diff
@@ -36,10 +37,35 @@ bool isInsideFlatpak()
+    static int supportedContainer = -1;
+
+    // Being in a container does not mean sub-containers cannot work. It depends upon various details such as
+    // docker vs podman, which permissions are given, is it privileged or unprivilged, and are unprivileged user namespaces enabled.
```
```suggestion
    // docker vs podman, which permissions are given, is it privileged or unprivileged, and are unprivileged user namespaces enabled.
```

mcatanzaro commented on Source/WTF/wtf/glib/Sandbox.cpp:
```diff
@@ -36,10 +37,35 @@ bool isInsideFlatpak()
+            &waitStatus, &error.outPtr());
+        supportedContainer = spawnSucceeded && g_spawn_check_wait_status(waitStatus, &error.outPtr());
+        if (!supportedContainer)
+            g_warning("Bubblewrap does not work inside of this container, sandboxing will be disabled: %s", error->message);
```
Too noisy. We shouldn't warn about this as it's expected and not an error. We should do at most a debug log here.

TingPing replied:
I'd really like to keep the warning in some form that it is disabling the sandbox. We are running webkit in podman on various distributions where it could work on one place and not another and a silent failure is counter productive.

mcatanzaro replied:
OK... try using WTFLogAlways then. It'll still be noisy, but at least somewhat less so than g_warning().

TingPing committed:
[GLib] Re-enable bwrap sandbox in containers when supported
https://bugs.webkit.org/show_bug.cgi?id=256917

Reviewed by Michael Catanzaro.

This detects if bwrap actually works inside of a container instead
of always disabling it.

* Source/WTF/wtf/glib/Sandbox.cpp:
(WTF::isInsideUnsupportedContainer):
(WTF::isInsideContainer): Deleted.
* Source/WTF/wtf/glib/Sandbox.h:
* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):

Canonical link: https://commits.webkit.org/264196@main
